dropwizard-jersey/src/main/java/io/dropwizard/jersey/params/AbstractParamConverterProvider.java

```diff
@@ -0,0 +1,60 @@
+package io.dropwizard.jersey.params;
+
+import com.google.common.base.Strings;
+import io.dropwizard.jersey.validation.JerseyParameterNameProvider;
+
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+
+public class AbstractParamConverterProvider implements ParamConverterProvider {
+
+    public AbstractParamConverterProvider() {
+    }
+
+    @Override
+    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
+        if (AbstractParam.class.isAssignableFrom(rawType)) {
+            final String parameterName = JerseyParameterNameProvider.getParameterNameFromAnnotations(annotations).orElse("Parameter");
+            final Constructor<T> constructor;
+            try {
+                constructor = rawType.getConstructor(String.class, String.class);
+            } catch (NoSuchMethodException ignored) {
+                // The Param class did not have a (String, String) constructor. We return null,
+                // leaving Jersey to handle these parameters as it normally would.
+                return null;
+            }
+            return new ParamConverter<T>() {
```

Comment:
Do you know how it will break backwards compatibility because doesn't the current classes get instantiated through [`StringConstructor`](https://github.com/jersey/jersey/blob/927b4d0605aeb119d15e9623dab65c8aec2c8e20/core-server/src/main/java/org/glassfish/jersey/server/internal/inject/ParamConverters.java#L118-L137), which calls [`AbstractStringReader`](https://github.com/jersey/jersey/blob/927b4d0605aeb119d15e9623dab65c8aec2c8e20/core-server/src/main/java/org/glassfish/jersey/server/internal/inject/ParamConverters.java#L77-L112)